CalorieTracker/CalorieTracker/CalorieTrackerViewController.swift

```diff
@@ -0,0 +1,188 @@
+        }
+    }
+    //MARK: - IBACTION -
+    @IBAction func addButtonTapped(_ sender: Any) {
+        // Create alert
+        let alert = UIAlertController(title: "Add calorie intake",
+                                      message: "Enter the amount of calories",
+                                      preferredStyle: .alert)
+
+        // Add text field
+        alert.addTextField { textField in
+            textField.placeholder = "Calories"
+        }
+        alert.addAction(UIAlertAction(title: "Save", style: .default, handler: { _ in
+            if alert.textFields?.first?.text?.isNumeric == false {
+                self.didNotEnterCaloriesAlert()
+                print("Empty")
+            }
+            guard let currentTextField = alert.textFields,
+                let caloriesString = currentTextField[0].text,
+                let calories = Int(caloriesString)
+                else { return }
+            // Create calories from given integer.
+            self.calorieController.add(amount: calories)
+        }))
+        alert.addAction(UIAlertAction(title: "Cancel",
+                                      style: .destructive,
+                                      handler: nil))
+        present(alert, animated: true, completion: nil)
+    }
+    func didNotEnterCaloriesAlert() {
+        let alert = UIAlertController(title: "Error Adding Calories",
+                                      message: "Need to enter a number.",
+                                      preferredStyle: .alert)
+        alert.addAction(UIAlertAction(title: "OK",
+                                      style: .default,
+                                      handler: nil))
+        present(alert, animated: true, completion: nil)
+    }
+}
+
+extension CalorieTrackerViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle,forRowAt indexPath:IndexPath) {
+        if editingStyle == .delete {
+            let calorie = fetchedResultsController.object(at: indexPath)
+            calorieController.delete(calorie: calorie) { _ in
+                DispatchQueue.main.async {
+                    self.tableView.reloadData()
+                }
+            }
+        }
+    }
+}
+
+extension CalorieTrackerViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return fetchedResultsController.sections?[section].numberOfObjects ?? 0
+    }
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: .calorieCell, for: indexPath)
+        let calorie = fetchedResultsController.object(at: indexPath)
+        guard let timestamp = calorie.timestamp else {
+            return UITableViewCell()
+        }
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "MMMM dd, yyyy h:mm a"
```

Comment:
Date formatters are kinda expensive to make, so this would be better to be a property. Making it a lazy var that is instantiated with a closure would work well.